vyper/ast/folding.py

```diff
@@ -176,16 +176,36 @@ def replace_user_defined_constants(vyper_module: vy_ast.Module) -> int:
 
         # Extract type definition from propagated annotation
         constant_annotation = node.get("annotation.args")[0]
-        type_ = (
-            get_type_from_annotation(constant_annotation, DataLocation.UNSET)
-            if constant_annotation
-            else None
-        )
+        try:
+            type_ = (
+                get_type_from_annotation(constant_annotation, DataLocation.UNSET)
+                if constant_annotation
+                else None
+            )
+        except UnknownType:
+            # Handle structs as user-defined types
```

Comment:
```suggestion
            # handle user-defined types e.g. structs - it's OK to not
            # propagate the type annotation here because user-defined
            # types can be unambiguously inferred at typechecking time
```